core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,151 +19,360 @@
+            this.settings.analyzeWildcard(DEFAULT_ANALYZE_WILDCARD);
+        }
         return this;
     }
 
+    /** Returns whether wildcards should by analyzed. */
+    public boolean analyzeWildcard() {
+        return this.settings.analyzeWildcard();
+    }
+
+    /**
+     * Specifies the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
     public SimpleQueryStringBuilder minimumShouldMatch(String minimumShouldMatch) {
         this.minimumShouldMatch = minimumShouldMatch;
         return this;
     }
 
+    /**
+     * Returns the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
+    public String minimumShouldMatch() {
+        return minimumShouldMatch;
+    }
+
+    /** 
+     * {@inheritDoc} 
+     * 
+     * Checks that mandatory queryText is neither null nor empty.
+     * */
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+
+        // Query text is required
+        if (queryText == null) {
+            validationException = QueryValidationException.addValidationError("[" + SimpleQueryStringBuilder.NAME + "] query text missing",
+                    validationException);
+        }
+
+        return validationException;
+    }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        // Use the default field (_all) if no fields specified
+        if (fieldsAndWeights.isEmpty()) {
+            String field = parseContext.defaultField();
+            fieldsAndWeights.put(field, 1.0F);
+        }
+
+        // Use standard analyzer by default if none specified
+        Analyzer luceneAnalyzer;
+        if (analyzer == null) {
+            luceneAnalyzer = parseContext.mapperService().searchAnalyzer();
+        } else {
+            luceneAnalyzer = parseContext.analysisService().analyzer(analyzer);
+            if (luceneAnalyzer == null) {
+                throw new QueryParsingException(parseContext, "[" + SimpleQueryStringBuilder.NAME + "] analyzer [" + analyzer
+                        + "] not found");
+            }
+
+        }
+        SimpleQueryParser sqp = new SimpleQueryParser(luceneAnalyzer, fieldsAndWeights, flags, settings);
+
+        if (defaultOperator != null) {
+            switch (defaultOperator) {
+                case OR:
+                    sqp.setDefaultOperator(Occur.SHOULD);
+                    break;
+                case AND:
+                    sqp.setDefaultOperator(Occur.MUST);
+                    break;
+            }
+        }
+
+        Query query = sqp.parse(queryText);
+        if (queryName != null) {
+            parseContext.addNamedQuery(queryName, query);
+        }
+
+        if (minimumShouldMatch != null && query instanceof BooleanQuery) {
+            Queries.applyMinimumShouldMatch((BooleanQuery) query, minimumShouldMatch);
+        }
+        query.setBoost(boost);
```

Comment:
http://svn.apache.org/viewvc/lucene/dev/trunk/lucene/queryparser/src/java/org/apache/lucene/queryparser/simple/SimpleQueryParser.java?view=markup line 150 and following makes me believe this cannot be null. @dakrone do you have time to double check?